tests/models/test_simple_nn.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Sequence
+
+import jax
+import jax.numpy as jnp
+import pytest
+from infra.module_tester import ComparisonMetric, test, test_with_random_inputs
+from infra.test_model import TestModel
+
+
+class SimpleNN(TestModel):  # TODO what's the benefit of inheriting nnx.Module?
```

Comment:
@mrakitaTT per [flax docs](https://flax.readthedocs.io/en/latest/api_reference/flax.nnx/module.html) I saw no reason to inherit from `nnx.Module`. It runs just fine as is now. Inheriting from `nnx.Module`, if I understood correctly, can mainly help with nested submodules and training mode. We have none of those for now.

This inheritance, if needed, could be provided through inheritance chain by letting `TestModel` inherit from `nnx.Module`, or concrete test model classes can use multiple inheritance since `TestModel` is only an interface class.

Reply:
We definitely don't want our ModelTester to inherit that, nor use ModuleTester as an ML module. ML module should be created and returned from the `load_model` function. That will usually happen by loading model from some repo like hugging face, but for the models that we manually make (for example by finding code on github or in ML paper) we should have a separate file implementing the model, and that class which implements the model might inherit from `nnx.Module` if we use flax to implement the model.